overlord/snapshotstate/backend/backend.go

```diff
@@ -565,7 +565,11 @@ func Import(ctx context.Context, id uint64, r io.Reader) (snapNames []string, er
 	defer tr.Cancel()
 
 	// Unpack and validate the streamed data
-	snapNames, err = unpackVerifySnapshotImport(r, id)
+	//
+	// XXX: this will leak snapshot IDs, i.e. we allocate a new
+	// snapshot ID before but then we error here because of e.g.
+	// duplicated import attempts
```

Comment:
I don't think this is a problem, it's is just an internal counter and these numbers generally shouldn't bother end users. FWTW be do create automatic snapshots that expire after a while behind user's back, so "gaps" in the numbering will be there.

Reply:
instead of failing we also have the option to change import to return the old id and return that to the client, but we also need to reset any expiration if the old snapshot was automatic (ideally we should stop marking if a snapshot is automatic in the zip but just carry that info the state where we carry expirations as well)

Reply:
Done. I kept DuplicatedSnapshotImportError internally between backend and snapshotstate but it is not surfaced anymore; an alternative approach would be to return actual set id but slightly annoying (less readable) as it goes through 2 helpers (unpackVerifySnapshotImport and checkDuplicatedSnapshotSetWithContentHash).